aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1070,36 +1070,52 @@ Tensor index_select_sparse(const Tensor& self, int64_t dim, const Tensor& index)
   }
 }
 
-Tensor slice(const Tensor& self, int64_t dim, int64_t start, int64_t end, int64_t step) {
+Tensor slice(
+    const Tensor& self,
+    int64_t dim,
+    c10::optional<int64_t> start,
+    c10::optional<int64_t> end,
+    c10::optional<int64_t> step) {
   int64_t ndim = self.dim();
   if (ndim == 0) {
     TORCH_CHECK_INDEX(false, "slice() cannot be applied to a 0-dim tensor.");
   }
   dim = maybe_wrap_dim(dim, ndim);
   auto sizes = self.sizes().vec();
   auto strides = self.strides().vec();
+
+  // handle optinal parameters
+  int64_t start_val = start.has_value() ? start.value() : 0;
+  int64_t end_val = end.has_value() ? end.value() : INT64_MAX;
+  int64_t step_val = step.has_value() ? step.value() : 1;
+
   // TODO: support negative strides
-  TORCH_CHECK(step > 0, "slice step must be positive");
```

Comment:
Why were these lines changed ? I think we should leave negative indexing NYI for tensors, it might be tricky, and i think is out of scope for this PR. Just error'ing shoudl be fine.

Reply:
I think it was just clang formatting

Reply:
oh, nvm then my bad. Changes heree look good 👍